### PR TITLE
Using invariance culture when converting to string

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
related Issue:
- [Exception while auto-train with regression task](https://github.com/dotnet/machinelearning-modelbuilder/issues/363)
- #4634

### CODE REVIEW

File: src/Microsoft.ML.AutoML/Utils/SweepableParamAttributes.cs [hunk 1]
```diff
@@ -145,7 +145,7 @@ public SweepableFloatParam(float min, float max, float stepSize = -1, int numSte
 
         public override void SetUsingValueText(string valueText)
         {
-            RawValue = float.Parse(valueText, CultureInfo.InvariantCulture);
+            RawValue = float.Parse(valueText);
```
Comment (justinormont) [inline src/Microsoft.ML.AutoML/Utils/SweepableParamAttributes.cs]: I would instead push the culture invariant value into the sweepable parameter:
https://github.com/dotnet/machinelearning/blob/b861b5d64841cbe0f2c866ee7586872aac450a51/src/Microsoft.ML.Sweeper/Parameters.cs#L140

**Comment by justinormont** [comment]
Closing/reopening to restart CI tests. 

CI is failing w/ an odd Git checkout issue: 

> 2020-01-21T17:01:18.2670411Z ##[command]git checkout --progress --force 36a6ca34b8cfdbf16af12493a49e40fc6800a32e
2020-01-21T17:01:18.2671965Z fatal: reference is not a tree: 36a6ca34b8cfdbf16af12493a49e40fc6800a32e
2020-01-21T17:01:18.2722749Z ##[error]Git checkout failed with exit code: 128

https://dev.azure.com/dnceng/public/_build/results?buildId=477842&view=logs&j=2302d3fe-129a-52d2-3973-322aaec0c902&t=12a19c0e-c998-51b7-645f-93b51cf4ff66&l=291

It sounds like it its trying to read from an non-existent commit hash. Perhaps restarting the CI will get the CI to read the current commit ID.

**Comment by sharwell** [comment]
Requesting a review from @tannergooding

**Comment by tannergooding** [comment]
This will break `+/-Infinity` as Invariant Culture is `Infinity` while other cultures may use things like `Inf` or `∞` (the latter is the default for `en-us`).

It is also likely to break users who expect or depend on the invariant culture working (a culture which uses `,` rather than `.` but gets `0.1` as the input may fail to parse or give an unexpected result).

I would recommend just updating the `ToString` call to also print using `InvariantCulture` as @justinormont suggested here: https://github.com/dotnet/machinelearning/pull/4635#discussion_r363992805

**Comment by tannergooding** [comment]
Ah, nevermind. I see that the title no longer matches the change being provided. The actual change is updating `ToString` which should work as expected. :+1:

**Comment by justinormont** [comment]
For the required `MachineLearning-CI`, one unit test is failing (randomly). The current unstable one is a TensorFlow test (`TensorFlowImageClassificationDefault`):
```
2020-01-21T18:41:12.4604444Z Starting test: Microsoft.ML.Scenarios.TensorFlowScenariosTests.TensorFlowImageClassificationDefault
2020-01-21T18:41:13.0889399Z   X Microsoft.ML.Scenarios.TensorFlowScenariosTests.TensorFlowImageClassificationWithPolynomialLRScheduling [17s 474ms]
2020-01-21T18:41:13.3391570Z   Error Message:
2020-01-21T18:41:13.3878708Z    System.Runtime.InteropServices.SEHException : External component has thrown an exception.
2020-01-21T18:41:13.3881327Z   Stack Trace:
2020-01-21T18:41:13.4477236Z      at Tensorflow.c_api.TF_SessionRun(IntPtr session, TF_Buffer* run_options, TF_Output[] inputs, IntPtr[] input_values, Int32 ninputs, TF_Output[] outputs, IntPtr[] output_values, Int32 noutputs, IntPtr[] target_opers, Int32 ntargets, IntPtr run_metadata, IntPtr status)
2020-01-21T18:41:13.4477710Z    at Microsoft.ML.TensorFlow.TensorFlowUtils.Runner.Run() in D:\a\1\s\src\Microsoft.ML.TensorFlow\TensorflowUtils.cs:line 491
2020-01-21T18:41:13.4478074Z    at Microsoft.ML.Vision.ImageClassificationTrainer.TrainAndEvaluateClassificationLayerCore(Int32 epoch, Single learningRate, Int32 featureFileStartOffset, ImageClassificationMetrics metrics, Int64[] labelTensorShape, Int64[] featureTensorShape, Int32 batchSize, Stream trainSetLabelReader, Stream trainSetFeatureReader, Byte[] labelBufferBytes, Byte[] featuresBufferBytes, Int32 labelBufferSizeInBytes, Int32 featureBufferSizeInBytes, Int32 featureFileRecordSize, LearningRateScheduler learningRateScheduler, DnnTrainState trainState, Runner runner, IntPtr featureBufferPtr, IntPtr labelBufferPtr, Action`2 metricsAggregator) in D:\a\1\s\src\Microsoft.ML.Vision\ImageClassificationTrainer.cs:line 1115
2020-01-21T18:41:13.4478827Z    at Microsoft.ML.Vision.ImageClassificationTrainer.TrainAndEvaluateClassificationLayer(String trainBottleneckFilePath, String validationSetBottleneckFilePath) in D:\a\1\s\src\Microsoft.ML.Vision\ImageClassificationTrainer.cs:line 999
2020-01-21T18:41:13.4479684Z    at Microsoft.ML.Vision.ImageClassificationTrainer.TrainModelCore(TrainContext trainContext) in D:\a\1\s\src\Microsoft.ML.Vision\ImageClassificationTrainer.cs:line 710
2020-01-21T18:41:13.4480104Z    at Microsoft.ML.Trainers.TrainerEstimatorBase`2.TrainTransformer(IDataView trainSet, IDataView validationSet, IPredictor initPredictor) in D:\a\1\s\src\Microsoft.ML.Data\Training\TrainerEstimatorBase.cs:line 157
2020-01-21T18:41:13.4480343Z    at Microsoft.ML.Trainers.TrainerEstimatorBase`2.Fit(IDataView input) in D:\a\1\s\src\Microsoft.ML.Data\Training\TrainerEstimatorBase.cs:line 77
2020-01-21T18:41:13.4480554Z    at Microsoft.ML.Data.EstimatorChain`1.Fit(IDataView input) in D:\a\1\s\src\Microsoft.ML.Data\DataLoadSave\EstimatorChain.cs:line 67
2020-01-21T18:41:13.4480901Z    at Microsoft.ML.Scenarios.TensorFlowScenariosTests.TensorFlowImageClassificationWithLRScheduling(LearningRateScheduler learningRateScheduler, Int32 epoch) in D:\a\1\s\test\Microsoft.ML.Tests\ScenariosWithDirectInstantiation\TensorflowTests.cs:line 1544
2020-01-21T18:41:13.4481204Z    at Microsoft.ML.Scenarios.TensorFlowScenariosTests.TensorFlowImageClassificationWithPolynomialLRScheduling() in D:\a\1\s\test\Microsoft.ML.Tests\ScenariosWithDirectInstantiation\TensorflowTests.cs:line 1474
```
https://dev.azure.com/dnceng/public/_build/results?buildId=491199&view=logs&j=dd8eddb6-ecc6-5f65-73e6-df90e5693b94

It's only failing under `Windows_x64_NetCoreApp21 Debug_Build`, so it's unlikely to be deterministic. 

I'll rerun the failing CI leg and hopefully the randomness falls in our favor. The previous Git issue was deterministic and restarting the CI seems to cause it to pick up the right commit ID.

**Pull request merged**